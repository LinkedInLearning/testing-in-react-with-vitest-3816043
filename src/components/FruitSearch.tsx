import { useEffect, useState } from 'react';
import { AppDispatch, RootState } from '../app/store';
import { useDispatch, useSelector } from 'react-redux';
import { fetchFruit } from '../features/fruits/fruitSlice';

const FruitSearch = () => {
  const [fruitName, setFruitName] = useState('');

  const dispatch: AppDispatch = useDispatch();
  
  const fruit = useSelector((state: RootState) => state.fruits.fruit);
  const status = useSelector((state: RootState) => state.fruits.status);
  const error = useSelector((state: RootState) => state.fruits.error);


  useEffect(() => {
    if (fruitName) {
      dispatch(fetchFruit(fruitName))
    }

    return () => {};
  }, [fruitName]);

  const handleInputChange = (event: {
    target: { value: string };
  }) => {
    setFruitName(event.target.value);
  };

  return (
    <div>
      <div className="row mb-5 text-center justify-content-center">
        <div className="col-lg-4 col-lg-offset-4">
          <div className="container input-group">
            <input
              className="form-control"
              type="text"
              value={fruitName}
              onChange={handleInputChange}
              placeholder="Search for a fruit"
            />
          </div>
        </div>
      </div>
      {status === 'loading' && (
        <div className="container text-center justify-content-center card p-3">
          <div className="card-body">
            <p className="card-text">Loading...</p>
          </div>
        </div>
      )}

      {status === 'failed' && error && (
        <div className="justify-content-center card p-3">{String(error)}</div>
      )}

      {status === 'succeeded' && fruit ? (
        <div
          className="container text-center justify-content-center card p-3"
          key={fruit.id}
        >
          <div className="card-body">
            <h1 className="card-title">Fruit: {fruit.name}</h1>
            <p className="card-subtitle">Family: {fruit.family}</p>
            <p className="card-text">Genus: {fruit.genus}</p>
            <p className="card-text">Order: {fruit.order}</p>
            <p className="card-text">Calories: {fruit.nutritions.calories}</p>
            <p className="card-text">Fat: {fruit.nutritions.fat}</p>
            <p className="card-text">Sugar: {fruit.nutritions.sugar}</p>
            <p className="card-text">
              Carbohydrates: {fruit.nutritions.carbohydrates}
            </p>
            <p className="card-text">Protein: {fruit.nutritions.protein}</p>
          </div>
        </div>
      ) : (
        <div className="container text-center justify-content-center card p-3">
          <div className="card-body">
            <p className="card-text">No fruit found {fruitName}</p>
          </div>
        </div>
      )}
    </div>
  );
};

export default FruitSearch;
