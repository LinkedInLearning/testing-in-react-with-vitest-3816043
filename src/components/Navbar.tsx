import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';

function NavBar() {
  return (
    <Navbar expand="lg" className="bg-body-tertiary">
      <Container>
        <Navbar.Brand href="#home">Testing in React</Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="me-auto justify-content-end">
            <Nav.Link href="#home">Home</Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
}

export default NavBar;
