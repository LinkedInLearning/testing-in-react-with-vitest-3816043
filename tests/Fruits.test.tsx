import { describe, expect, it } from 'vitest';

import { Fruits } from '../src/components/Fruits';

describe('Fruit Snapshot Test', () => {
  it('Should load fruit data', async () => {
    // ARRANGE
    const fruit = await Fruits('strawberry');

    // ASSERT
    expect(fruit).toMatchInlineSnapshot(`
    {
      "family": "Rosaceae",
      "genus": "Fragaria",
      "id": 1,
      "name": "strawberry",
      "nutritions": {
        "calories": 29,
        "carbohydrates": 5.5,
        "fat": 0.4,
        "protein": 0.8,
        "sugar": 5.4,
      },
      "order": "Rosales",
    }
    `);
  });

  it('Should create snapshots file', async () => {
    // ARRANGE
    const fruit = await Fruits('strawberry');

    // ASSERT
    expect(fruit).toMatchSnapshot();
  });
});
