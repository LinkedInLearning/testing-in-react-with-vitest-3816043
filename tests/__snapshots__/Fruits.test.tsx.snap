// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Fruit Snapshot Test > Should create snapshots file 1`] = `
{
  "family": "Rosaceae",
  "genus": "Fragaria",
  "id": 1,
  "name": "strawberry",
  "nutritions": {
    "calories": 29,
    "carbohydrates": 5.5,
    "fat": 0.4,
    "protein": 0.8,
    "sugar": 5.4,
  },
  "order": "Rosales",
}
`;
